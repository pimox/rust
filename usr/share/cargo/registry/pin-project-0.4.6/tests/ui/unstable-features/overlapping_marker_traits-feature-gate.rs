// NB: If you change this test, change 'overlapping_marker_traits.rs' at the same time.

use pin_project::pin_project;
use std::marker::PhantomPinned;

#[pin_project] //~ ERROR E0119
struct Foo<T> {
    #[pin]
    x: T,
}

// unsound Unpin impl
impl<T> Unpin for Foo<T> {}

fn is_unpin<T: Unpin>() {}

fn main() {
    is_unpin::<Foo<PhantomPinned>>()
}
