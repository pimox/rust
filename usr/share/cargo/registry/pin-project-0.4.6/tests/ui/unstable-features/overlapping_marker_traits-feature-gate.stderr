error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `Foo<_>`:
  --> $DIR/overlapping_marker_traits-feature-gate.rs:6:1
   |
6  | #[pin_project] //~ ERROR E0119
   | ^^^^^^^^^^^^^^ conflicting implementation for `Foo<_>`
...
13 | impl<T> Unpin for Foo<T> {}
   | ------------------------ first implementation here
